<a href="https://colab.research.google.com/github/rkrissada/100DayOfMLCode/blob/master/day_066_predict_employess_turnover.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Predict Employess Turnover

## Data Preprocessing

In [4]:
import pandas as pd

hr = pd.read_csv('HR_comma_sep.csv')
col_names = hr.columns.tolist()
print("Column names:")
print(col_names)

print("Sample data:")
hr.head()

Column names:
['satisfaction_level', 'last_evaluation', 'number_project', 'average_montly_hours', 'time_spend_company', 'Work_accident', 'left', 'promotion_last_5years', 'sales', 'salary']
Sample data:


satisfaction_level  last_evaluation  number_project  average_montly_hours  \
0                0.38             0.53               2                   157   
1                0.80             0.86               5                   262   
2                0.11             0.88               7                   272   
3                0.72             0.87               5                   223   
4                0.37             0.52               2                   159   

   time_spend_company  Work_accident  left  promotion_last_5years  sales  \
0                   3              0     1                      0  sales   
1                   6              0     1                      0  sales   
2                   4              0     1                      0  sales   
3                   5              0     1                      0  sales   
4                   3              0     1                      0  sales   

   salary  
0     low  
1  medium  
2  medium  
3     low  
4     low

Rename column name from "sales" to "department"

In [0]:
hr=hr.rename(columns = {'sales':'department'})

The type of the columns can be found out as follows:

In [6]:
hr.dtypes

satisfaction_level       float64
last_evaluation          float64
number_project             int64
average_montly_hours       int64
time_spend_company         int64
Work_accident              int64
left                       int64
promotion_last_5years      int64
department                object
salary                    object
dtype: object

In [7]:
hr.isnull().any()

satisfaction_level       False
last_evaluation          False
number_project           False
average_montly_hours     False
time_spend_company       False
Work_accident            False
left                     False
promotion_last_5years    False
department               False
salary                   False
dtype: bool

Our data is prett clean, no missing value

In [8]:
hr.shape

(14999, 10)

The data contains 14,999 employees and 10 features.

The "left" column is the outcome variable recoding 1 and 0. 1 for employee who left the company and 0 for those who didn't.

The department column of the dataset has many categories and we need to reduce the categories for a better modeling. The department column has the following categories:

In [9]:
hr['department'].unique()

array(['sales', 'accounting', 'hr', 'technical', 'support', 'management',
       'IT', 'product_mng', 'marketing', 'RandD'], dtype=object)

Let us combine "technical","support" and "IT" together and call them "technical".

In [0]:
import numpy as np

hr['department'] = np.where(hr['department'] == 'support','technical',hr['department'])
hr['department'] = np.where(hr['department'] == 'IT','technical',hr['department'])

After the change, this is how the department categories look:

In [12]:
hr['department'].unique()

array(['sales', 'accounting', 'hr', 'technical', 'management',
       'product_mng', 'marketing', 'RandD'], dtype=object)